File: my_py_pkg/my_py_pkg/robot_news_station.py
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

class RobotNewsStationNode(Node):
    def __init__(self):
        super().__init__("robot_news_station")
        self.declare_parameter("robot_name", "Sushant_bot")
        self.publisher_=self.create_publisher(String,"robot_news", 10)
        self.timer_= self.create_timer(0.5,self.publish_news)
        self.robot_name_=self.get_parameter("robot_name")
        self.get_logger().info(f"Hi this is {self.robot_name_} ")
        self.count=0
    def publish_news(self):
        msg= String()
        self.count+=1
        msg.data = "I hope you are all well right !... " + str(self.count)
        self.publisher_.publish(msg)


def main(args=None):
    rclpy.init(args=args)
    node=RobotNewsStationNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__=="__main__":
    main()

File: ros2_ws/src/my_cpp_pkg/src/robot_news_station.cpp
#include <rclcpp/rclcpp.hpp>
#include <example_interfaces/msg/string.hpp>

using namespace std::chrono_literals;

class RobotNewsStation : public rclcpp::Node //This class inherits from rclcpp::Node
{
    public:
    RobotNewsStation() : Node("robot_news_station"), robot_name_("ROS2_Robot_Sushant")
    {
        publisher_= this->create_publisher<example_interfaces::msg::String>("robot_news",10);
        timer_ = this->create_wall_timer(0.5s, std::bind(&RobotNewsStation::publishNews, this));
        RCLCPP_INFO(this->get_logger(), "Robot_News_Station node has been Started");
    }

    private:
    void publishNews(){
        auto msg= example_interfaces::msg::String();
        msg.data = std::string("Hello, this is ") + robot_name_ + std::string(" From the robot news station.");
        publisher_->publish(msg);
    }
        std::string robot_name_;
        rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
        rclcpp::TimerBase::SharedPtr timer_;
};
    
int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<RobotNewsStation>());
    rclcpp::shutdown();
    return 0;
}